doc/adr/0008-TimeRangeData-widget.md

```diff
@@ -0,0 +1,53 @@
+# 8. Time Range Data Widget
+
+Date: 2023-1-10
+
+## Status
+
+New
+
+## Context
+
+There has been a request for the ability for users to be able to select part of a trace by copy/pasting timestamp data.  An attempt was made to implement this feature, but a consensus could not be reached on the UI/UX.  Design needs to be agreed upon before any further implementation.
+
+The proposed design is subject to change based on feedback.
+
+### Proposed UX Design
+
+#### UX Flow
+
+A "Time Range Data" widget will be added to the left panel.
+
+![current](0008/widget-sidebar.png)
+
+The widget will display: View Start, View End, Selection Start, Selection End, and an Edit button.
+
+![current](0008/time-range-data-widget-1.png)
+
+Clicking on the edit button turns the display into a form.  Only the selection start/end will be editable.
+
+![current](0008/time-range-data-widget-2.png)
+
+If the selected time is invalid, the user will be notified onSubmit with a red asterisk and warning message at the top.  Submit and Cancel buttons will remain visible.
+
+![current](0008/time-range-data-widget-3.png)
+
+If the selected time is valid, the selection range will be set and the widget will return to it's initial state.
+
+#### UX Miscellaneous Behaviors
+* The view range will not change when the selection range is changed.  This includes panning and zooming.
+* The time will be displayed in absolute time as nanoseconds, since that is the format users timestamp data will be.
```

Comment:
can we have a push to view range, and 'follow selection'?

Reply:
Maybe we can have a dialog popup for the user to confirm, something like `Follow selection`, or even better, a icon button to quickly move to the selection range - this might be useful to help users to navigate quickly back to the selection range in case they moved elsewhere, or did not follow the selection initially.

Reply:
IMO pop-ups should be avoided, as they can get annoying after seeing them numerous times.

Moving the view range based on selection range has been an ongoing discussion.  When it comes to view-range changing based on selection-range in the theia-trace-extension, the behavior is mixed. An application wide solution can be implemented in the future.  For example: a toggleable setting for "follow selection", or a button that centers the selection in the view range.

I think the topic needs a separate discussion and should not bottleneck the implementation of the widget.  Not moving the view range is the general behavior of the application, we can change it in the widget after a consensus is reached.

Reply:
This sounds good, I think an application wide feature for this will benefit the user more. Let's keep this topic for further discussion. We should backlog this somewhere if we have not done this already.